### PR TITLE
add vscode extensions & settings

### PR DESCRIPTION
**Pull request by KATT**
This is a bit of a controversial topic whether or not to have editor settings checked in. I like it, but everyone doesn't.

Will comment inline on what each part does.

### CODE REVIEW

File: .vscode/extensions.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "recommendations": [
+    "yzhang.markdown-all-in-one",
```
Comment (KATT) [inline .vscode/extensions.json:4]: Markdown plugin to make it easier to use `.md`

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+    "esbenp.prettier-vscode",
```
Comment (KATT) [inline .vscode/extensions.json:6]: Add prettier

File: .vscode/extensions.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+    "dbaeumer.vscode-eslint",
```
Comment (KATT) [inline .vscode/extensions.json]: Add eslint

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+  "typescript.tsdk": "node_modules/typescript/lib",
```
Comment (KATT) [inline .vscode/settings.json:2]: Ensure we use the project's typescript version

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+  "editor.formatOnSave": true,
+  // Auto-fix issues with ESLint when you save code changes
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
+  },
+  "eslint.run": "onSave"
```
Comment (KATT) [inline .vscode/settings.json:8]: Format + fix linting on save when you save

**Comment by emrysal** [comment]
I wonder if it is possible to use https://github.com/calendso/calendso/blob/main/.editorconfig for this. After all https://editorconfig.org/ is explicitly supported by vscode (and I believe they are contributors).

https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2019

**Comment by KATT** [comment]
> I wonder if it is possible to use https://github.com/calendso/calendso/blob/main/.editorconfig for this. After all https://editorconfig.org/ is explicitly supported by vscode (and I believe they are contributors).
> 
> https://docs.microsoft.com/en-us/visualstudio/ide/create-portable-custom-editor-options?view=vs-2019

Not for all of it by any means. I.e. the tailwind sorter is super nice to have but I don't think editorconfig/prettier/eslint has something like it [yet]

**Comment by emrysal** [comment]
What do you think @baileypumfleet

**Comment by LecrisUT** [comment]
Please also include an appropriate `.gitignore` if this is to be added. Here is the suggestion generated by the Jetbrains plugin:
```
### VisualStudioCode template
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json
*.code-workspace

# Local History for Visual Studio Code
.history/
```

**Review by LecrisUT — CHANGES_REQUESTED** [review]
Please also include an appropriate `.gitignore` if this is to be added. Here is the suggestion generated by the Jetbrains plugin:
```
### VisualStudioCode template
.vscode/*
!.vscode/settings.json
!.vscode/tasks.json
!.vscode/launch.json
!.vscode/extensions.json
*.code-workspace

# Local History for Visual Studio Code
.history/
```

**Comment by KATT** [comment]
Replaced by #485

**Pull request closed without merging**